webview_test.cc

```diff
@@ -83,11 +83,35 @@ static void test_bidir_comms() {
 // =================================================================
 static void test_json() {
   auto J = webview::json_parse;
+  // Valid input with expected output
   assert(J(R"({"foo":"bar"})", "foo", -1) == "bar");
   assert(J(R"({"foo":""})", "foo", -1) == "");
+  assert(J(R"({"foo":{}")", "foo", -1) == "{}");
   assert(J(R"({"foo": {"bar": 1}})", "foo", -1) == R"({"bar": 1})");
   assert(J(R"(["foo", "bar", "baz"])", "", 0) == "foo");
   assert(J(R"(["foo", "bar", "baz"])", "", 2) == "baz");
+  // Valid UTF-8 with expected output
+  assert(J(R"({"フー":"バー"})", "フー", -1) == "バー");
+  assert(J(R"(["フー", "バー", "バズ"])", "", 2) == "バズ");
+  // Invalid input with valid output - should probably fail
+  assert(J(R"({"foo":"bar")", "foo", -1) == "bar");
```

Comment:
I would expect failure here but it succeeds and returns a usable value.

Reply:
The usable value is correct, so I don't think this is a big problem unless it has unpredictable effects. I think we can fix it, if it doesn't make the implementation more complicated, but it doesn't seem to need to be high-priority.